main.tf

```diff
@@ -1,13 +1,89 @@
-resource "random_integer" "example" {
-  count = module.this.enabled ? 1 : 0
+#-----------------------------------------------------------------------------------------------------------------------
+# Subscribe the Acccount to GuardDuty
+#-----------------------------------------------------------------------------------------------------------------------
+resource "aws_guardduty_detector" "guardduty" {
+  enable                       = module.this.enabled
+  finding_publishing_frequency = var.finding_publishing_frequency
+}
+
+#-----------------------------------------------------------------------------------------------------------------------
+# Optionally configure Event Bridge Rules and SNS subscriptions 
+# https://docs.aws.amazon.com/securityhub/latest/userguide/securityhub-cwe-integration-types.html
+# https://docs.aws.amazon.com/AmazonCloudWatch/latest/events/resource-based-policies-cwe.html#sns-permissions
+#-----------------------------------------------------------------------------------------------------------------------
+module "sns_topic" {
+
+  source  = "cloudposse/sns-topic/aws"
+  version = "0.9.0"
+  count   = local.create_sns_topic ? 1 : 0
+
+  attributes      = ["guardduty"]
+  subscribers     = var.subscribers
+  sqs_dlq_enabled = false
+
+  context = module.this.context
+}
+
+module "findings_label" {
+  source  = "cloudposse/label/null"
+  version = "0.21.0"
+
+  attributes = ["guardduty-findings"]
+  context    = module.this.context
+}
+
+resource "aws_sns_topic_policy" "sns_topic_publish_policy" {
+  count  = module.this.enabled && local.create_sns_topic ? 1 : 0
+  arn    = local.findings_notification_arn
+  policy = data.aws_iam_policy_document.sns_topic_policy[0].json
+}
 
-  min = 1
-  max = 50000
-  keepers = {
-    example = var.example
+data "aws_iam_policy_document" "sns_topic_policy" {
+  count     = module.this.enabled && local.create_sns_topic ? 1 : 0
+  policy_id = "GuardDutyPublishToSNS"
+  statement {
+    sid = ""
+    actions = [
+      "sns:Publish"
+    ]
+    principals {
+      type        = "Service"
+      identifiers = ["cloudwatch.amazonaws.com"]
```

Comment:
not sure, but is this a correct identifier? ["cloudwatch.amazonaws.com"]

Reply:
I also thought it should be `events.amazonaws.com`, but after applying it was switched to `cloudwatch.amazonaws.com`, so there was a perpetual diff. I tested with the identifier set to `cloudwatch.amazonaws.com` and it worked (sent notifications to SNS).